### PR TITLE
Add automatic daily offers to admin CLI

### PR DESCRIPTION
**Pull request by delventhalz**
Adds two commands to the admin cli:

**Command:** `mktadm renew`

Similar to `seed` (formerly known as `submit`), `renew` takes a url and a data file and submits data to the REST API. Specifically, it will submit any `RENEWABLE` Offers, after first closing any open offers that match. This effectively resets rules like `EXCHANGE_ONCE_PER_ACCOUNT`.

**Usage:**

```
bin/mktadm renew -u localhost:8040 -d sawbuck_app/app_data.yaml
```

**Command:** `mktadm schedule`

Uses `cron` to execute other `mktadm` commands on a regular schedule. Can be run hourly, daily, weekly, or monthly. Resubmitting the same command (with the same parameters), will reschedule it. Can also use the `--remove`flag to unschedule a command.

**Usage**
```
bin/mktadm schedule -H "renew -u localhost:8040 -d sawbuck_app/app_data.yaml"
bin/mktadm schedule -D "renew -u localhost:8040 -d sawbuck_app/app_data.yaml"
bin/mktadm schedule --remove "renew -u localhost:8040 -d sawbuck_app/app_data.yaml"
```

You can check your existing cron jobs using:
```
crontab -l
```

Or remove them with:
```
crontab -r
```

### CODE REVIEW

File: admin/marketplace_admin/commands/schedule.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        'schedule_command',
+        type=str,
+        help='The mktadm command to schedule')
+
+    return parser
+
+
+def do_schedule(opts):
+    command = _get_command(opts)
+    crontab = subprocess.check_output(['crontab', '-l']).decode()
```
Comment (boydjohnson) [inline admin/marketplace_admin/commands/schedule.py]: You could catch the CalledProcessError that is raised here when `crontab -l` is run by a user that doesn't have a crontab set up, and set the crontab variable to the empty string.

**Review by boydjohnson — CHANGES_REQUESTED** [review]
schedule command assumes that user has crontabs already set.

**Review by boydjohnson — APPROVED** [review]
The reference finding mini-language is really neat.

File: admin/marketplace_admin/commands/renew.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def do_renew(opts):
+    LOGGER.info('Reading data: %s', opts.data)
+    renew_data = data.load(opts.data)
+
+    LOGGER.info('Submitting data to URL: %s', opts.url)
+    fetch = lambda path: api.get(opts.url, path)
+    submit = lambda p, b, a=None: api.post(opts.url, p, b, a)
+    update = lambda p, b, a: api.patch(opts.url, p, b, a)
```
Comment (peterschwarz) [inline admin/marketplace_admin/commands/renew.py]: Not exactly the pythonic way, but interesting. Looks `functools.partial` would solve this

Comment (delventhalz) [reply admin/marketplace_admin/commands/renew.py]: Being pythonic is no longer something I am particularly concerned with. Switched to partial.

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**